lib/bolt/config.rb

```diff
@@ -190,7 +193,10 @@ def self.load_bolt_yaml(dir)
 
       # Validate the config against the schema. This will raise a single error
       # with all validation errors.
-      Validator.new.validate(data, bolt_schema, filepath)
+      Validator.new.tap do |validator|
+        validator.validate(data, bolt_schema, filepath)
+        validator.warnings.each { |warning| logs << { warn: warning } }
```

Comment:
We actually want to add this to the `deprecations` list so that we get analytics on how many people see these warnings.